spec/models/community_users.rb

```diff
@@ -0,0 +1,84 @@
+require 'spec_helper'
+
+describe CommunityUsers do
+  let(:student) { create(:student) }
+  let(:student_looking_for_coach) { create(:student, :interested_in_coaching) }
+  let(:coach) { create(:coach, :available) }
+  describe '#all' do
+    before do
+      @params = {
+          role: '',
+          interest: '',
+          availability: '',
+          search: '',
+          location: '',
+          page: 1,
+          sort: '',
+          direction: 'asc'
+        }
+
+      @users = User.ordered(@params[:sort], @params[:direction])
+                   .group('users.id')
+                   .with_all_associations_joined
+
+      @community = CommunityUsers.new(@params, @users)
+    end
+
+    context 'return filtered users' do
+      describe 'with params presence' do
+        it 'return user with role specified' do
+          @params[:role] = student.roles.first.name
+
+          community = @community
+          expect(community.all).to match_array student
+        end
+
+        it 'return user with interested_in' do
+          @params[:interest] = student_looking_for_coach.interested_in
+
+          community = @community
+          expect(community.all).to match_array student_looking_for_coach
+        end
+
+        it 'return user (coach) with availability' do
+          @params[:availability] = coach.availability
+
+          community = @community
+          expect(community.all).to match_array coach
+        end
+
+        it 'return user by search' do
+          @params[:search] = student.name
+
+          community = @community
+          expect(community.all).to match_array student
+        end
+
+        it 'return user by location' do
+          @params[:location] = student.country
+
+          community = @community
+          expect(community.all).to match_array student
+        end
+      end
+    end
+
+    context 'private methods' do
```

Comment:
It's best practice to not test private methods. Unit tests would just cover the public interface and through this also indirectly test the internals.

Could you convert these tests so that you instead test the outcome of these scenarios on the `all` method?